magicctapipe/image/calib.py

```diff
@@ -0,0 +1,102 @@
+
+
+import numpy as np
+
+
+from ctapipe.image import (
+    apply_time_delta_cleaning,   
+    number_of_islands,
+    tailcuts_clean,   
+)
+
+from lstchain.image.cleaning import apply_dynamic_cleaning
+from lstchain.image.modifier import (
+    add_noise_in_pixels,
+    random_psf_smearer,    
+)
+
+__all__ = [
+    "Calibrate_LST", "Calibrate_MAGIC"
+]
+
+def Calibrate_LST(event, tel_id, rng, config_lst, camera_geoms, calibrator_lst, increase_nsb, use_time_delta_cleaning, use_dynamic_cleaning ):
```

Comment:
Not covered by test

Reply:
These functions are used in the mc_dl0_to_dl1 script to process data up to DL1, so it should be called (indirectly) while running this step in Git. But I can try to add a test, if needed. BTW, a few messages below Julian proposed to have one single Calibrate function instead of two, so we must first choose this (one or two functions) and then I can write the tests. In the meanwhile I have to study a bit these functions to implement easy and not CPU-intensive tests

Reply:
If the function is called in a test, even indirectly, it should be covered. So either the test skip this call (which is an issue if not expected and needs to be understood) or there is a bug in the coverage identification. In both cases a dedicated test could be useful. Of course changes to the code structure such as proposed by Julian come first (or actually better, in parallel).

Reply:
Closing because the function was generalized to work with both MAGIC and LST and because the tests were added.